numpy/add_newdocs.py

```diff
@@ -351,6 +353,7 @@ def outer_it(x, y, out=None):
             for (a, b, c) in it:
                 mulop(a, b, out=c)
 
+            it.close()
```

Comment:
This needs to be turned into a context manager, just add `with it:` and indent the for loop and return. Same for the `it.close()` that appear in a few other places in this file.

Also, I think you need `list(range(` in two places here. In python 3 ranges are generator objects that don't support `+`.

Reply:
fixed